### PR TITLE
Clean up cli

### PR DESCRIPTION
**Pull request by ghislainfourny**
This is a first step to clean up the Rumble execution. Now the code to execute locally or on a cluster is the same, which should reduce the risk of issues when releasing. 

All paths (absolute, relative, any system) now should work with --query-path and --log-path.

Also, the shell no longer writes the query to a temporary file.

### CODE REVIEW

File: src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java [hunk 1]
```diff
@@ -100,51 +102,22 @@ public void runLocal(String queryFile, String outputPath) throws IOException {
...
         if (result.isRDD() && outputPath != null) {
             JavaRDD<Item> rdd = result.getRDD(dynamicContext);
             JavaRDD<String> output = rdd.map(o -> o.serialize());
             output.saveAsTextFile(outputPath);
         } else {
             String output = runIterators(result, dynamicContext);
             if (outputPath != null) {
+                File outputFile = new File(outputPath);
                 List<String> lines = Arrays.asList(output);
                 org.apache.commons.io.FileUtils.writeLines(outputFile, "UTF-8", lines);
```
Comment (ghislainfourny) [inline src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java:115]: The plan is to also use the hadoop filesystem for writing back small output, similar to everything else, but I have not gotten to it yet.

File: src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java [hunk 2]
```diff
@@ -194,30 +167,6 @@ public String runInteractive(java.nio.file.Path queryFile) throws IOException {
         return rddOutput;
     }
 
-    private JsoniqLexer getInputSource(String arg) throws IOException {
-        arg = arg.trim();
-        // return embedded file
```
Comment (ghislainfourny) [inline src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java:199]: This was causing a lot of issues, it is now all handled in a unified way with the hadoop filesystem methods.

File: src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java [hunk 3]
```diff
@@ -21,73 +21,65 @@
...
+    private void checkOutputFile(String outputPath) throws IOException {
         if (outputPath != null) {
-            outputFile = new File(outputPath);
-            if (outputFile.exists()) {
+            if (UrlValidator.exists(outputPath, new ExceptionMetadata(0, 0))) {
                 if (!this.configuration.getOverwrite()) {
                     System.err.println(
                         "Output path " + outputPath + " already exists. Please use --overwrite yes to overwrite."
                     );
                     System.exit(1);
```
Comment (ghislainfourny) [inline src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java:66]: The Hadoop FileContext class will overwrite, no need to delete any more.

File: src/main/java/org/rumbledb/runtime/functions/input/FileSystemUtil.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new CannotRetrieveResourceException(
+                    "Error while accessing the " + locator.getScheme() + " filesystem.",
+                    metadata
+            );
+        }
+    }
+
+    public static boolean delete(String url, ExceptionMetadata metadata) {
```
Comment (ghislainfourny) [inline src/main/java/org/rumbledb/runtime/functions/input/FileSystemUtil.java:56]: Not actually needed any more, but might be in the future.

File: src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java [hunk 5]
```diff
@@ -108,79 +101,24 @@ public void runLocal(String queryFile, String outputPath) throws IOException {
...
-            JavaRDD<String> output = rdd.map(o -> o.serialize());
-            output.saveAsTextFile(outputPath);
-        }
         long endTime = System.currentTimeMillis();
         long totalTime = endTime - startTime;
-        if (this.configuration.getLogPath() != null) {
-            writeTimeLog(totalTime);
+        String logPath = this.configuration.getLogPath();
+        if (logPath != null) {
+            String time = "[ExecTime] " + totalTime;
```
Comment (CanBerker) [inline src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java:114]: The other execution time message has been refactored to:
`output("The query took " + time + " milliseconds to execute.");`

Was it intentional to leave this one as Exectime?

File: src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java (hunk 3 above)
Comment (CanBerker) [reply src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java:66]: Cleanup on this file looks amazing

**Pull request merged**